From 0ae46be1dd4151b7b5dc80a0271d5a86b8826af1 Mon Sep 17 00:00:00 2001
From: Tasos Sahanidis <tasos@tasossah.com>
Date: Tue, 12 May 2020 16:00:43 +0300
Subject: [PATCH] Don't enable keys we don't need in uinput

Specifically, enabling KEY_ALS_TOGGLE causes the kernel to complain
---
 src/daemon/input_linux.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/daemon/input_linux.c b/src/daemon/input_linux.c
index 8c8a604..10ffc92 100644
--- a/src/daemon/input_linux.c
+++ b/src/daemon/input_linux.c
@@ -28,9 +28,11 @@ int uinputopen(struct uinput_user_dev* indev, int mouse){
         for(int i = 0; i < REL_CNT; i++)
             ioctl(fd, UI_SET_RELBIT, i);
     } else {
-        // Enable all possible keys
-        for(int i = 0; i < KEY_CNT; i++)
+        // Enable common keyboard keys
+        for(int i = KEY_ESC; i <= KEY_MEDIA; i++)
             ioctl(fd, UI_SET_KEYBIT, i);
+        // Enable KEY_FN as that's also used
+        ioctl(fd, UI_SET_KEYBIT, KEY_FN);
         // Enable LEDs
         ioctl(fd, UI_SET_EVBIT, EV_LED);
         //for(int i = 0; i < LED_CNT; i++)
-- 
2.26.2

